### PR TITLE
added some new boards and universal header + blink samples

### PR DESCRIPTION
**Pull request by snovotill**
I have no idea how to use github but nevertheless here are some new board definitions (including Nuvoton) and examples which include the new MCU header files. The original 805x headers have also been made more universal by ensuring they contain pin definitions in both accepted formats, that being PXY as well as PX_Y. Extra Flags are added to the previously existing STC headers so that boards can be automatically detected at compile time as is demonstrated within anymcu-blink main.c

### CODE REVIEW

**Comment by snovotill** [comment]
Requesting some comments from the community. I'm currently working on BSP files (MCU header files) which need to be included for the STC and Nuvoton. I'll upload those soon.

**Comment by LovelyA72** [comment]
This is truly awesome! finally, we have nuvoton support in platformio. I don't know why still nobody is reviewing this pull request.

**Comment by cocus** [comment]
I've tried to use the 90C516RD+ but it's similar to the 89C516RD+ by the looks of it. This PR would've solved my issue beforehand... Not sure why it's not merged yet

File: boards/at89s51.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (ivankravets) [inline boards/at89s51.json:1]: Please RENAME all JSON files to the original TARGET NAMEs. Typically, make them UPPER CASE. We keep target devices in sync with vendor naming.

`boards/at89s51.json` -> `boards/AT89S51.json`

File: boards/eol.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (ivankravets) [inline boards/eol.sh:1]: Remove this file

File: examples/anymcu-blink/platformio.ini [hunk 3]
```diff
@@ -0,0 +1,13 @@
+; PlatformIO Project Configuration File
```
Comment (ivankravets) [inline examples/anymcu-blink/platformio.ini:1]: Please add new examples to the CI list
https://github.com/platformio/platform-intel_mcs51/blob/develop/.github/workflows/examples.yml#L12

File: examples/anymcu-blink/platformio.ini [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[env:n76e003]
```
Comment (ivankravets) [inline examples/anymcu-blink/platformio.ini:11]: It looks like this example supports multiple boards (target devices). Does it make sense to declare multiple configuration environments to test them?

File: examples/anymcu-blink/src/workspace.code-workspace [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (ivankravets) [inline examples/anymcu-blink/src/workspace.code-workspace:1]: Please remove redundant files.

**Review by ivankravets — CHANGES_REQUESTED** [review]
Thanks for the PR! Please resolve my comments above and we would be glad to merge this PR.

**Comment by ivankravets** [comment]
I fixed all issues, thanks for the PR! Anyone in this thread, please re-test this PR using the next guide:

1. Upgrade PIO Core to the latest dev version `pio upgrade --dev`
2. Use the GIT version of dev-platform in `platformio.ini` => `platform = https://github.com/platformio/platform-intel_mcs51.git`
3. Open the system terminal, change the working directory to the project folder, and type `pio pkg update`
4. Build project
5. Upload firmware.

Does it work now?

P.S: Please note that all board manifests were renamed to upper case.

**Comment by cocus** [comment]
Tested with a 90C516RD+, works as expected.

**Pull request merged**